OasisDEX/oasis-borrow
# Top assets and positions

# [Top assets and positions](https://app.shortcut.com/oazo-apps/story/3793/create-top-5-positions)
  
<img width="1330" alt="image" src="https://user-images.githubusercontent.com/3783198/169262671-39a105ac-88c7-496b-ac8b-635994df7152.png">

## Changes 👷‍♀️
- adds a new box for top assets and positions.  This includes stuff in the users wallet, as well their maker vaults (encoded as `Position`s)
- hides this box if there are no positions.
- assets open a menu that contains things people can do with an asset - clicking through takes them to the landing page for that asset
  
## Tech changes
- creates a new concept - position.  Not much in it right now, but the idea is Maker vaults can be exposed as positions.  Other concepts from other protocols can also be surfaced as positions in the future.  Feedback please.
- new tokenPriceUsd(tokens: Array<string>) pipe, using a new ticker from coingecko in addition to coinpaprika and coinbase.
- new subject in uichanges streams

## How to test 🧪
1. hardhat
2. enable `Earn` feature (UI is expected to be janky with the rest of the page - it will not get released like this)
3. grab some erc-20 assets using https://github.com/OasisDEX/multiply-proxy-actions/blob/develop/tasks/transfer-erc20.ts
4. play around with creating some positions from the positions overview panel
5. play around adjusting some positions.
6. check the piechart percentages
7. do this on mobile.
8. connected/not connected
9. Open swap widget from the positions list
